src/parse.rs

```diff
@@ -36,8 +36,15 @@ fn parse_a_hrefs(handle: &Handle, base_url: &Url, urls: &mut HashSet<Url>) {
                 .iter()
                 .find(|attr| &attr.name.local == "href")
             {
-                let val = &attr.value;
-                if let Ok(link) = base_url.join(val) {
+                let mut val = attr.value.clone();
+                // if the href value starts with / we need to join it to the base without the /
+                // otherwise with base_url = `foo`, val = `/bar`, base_url.join(val) == `/bar`
+                // and not `foo/bar`
+                if attr.value.starts_with('/') {
+                    val.pop_front_char();
+                }
```

Comment:
Hmm, so this is an interesting case because the path is absolute with respect to the _url_, not the file path. I think it's reasonable to treat it that way, the comment is not exactly how I'd describe it though:
```suggestion
                // Treat absolute paths as absolute with respect to the `base_url`, not with respect to the file system.
                if attr.value.starts_with('/') {
                    val.pop_front_char();
                }
```